FreeRTOS-Plus/Demo/AWS/Device_Defender_Windows_Simulator/Device_Defender_Demo/report_builder.c

```diff
@@ -323,6 +379,185 @@ static eReportBuilderStatus prvWriteConnectionsArray( char * pcBuffer,
+                                                    ( long long ) ( pxCustomMetric->xData.pllNumberList )[ i ] );
+                }
+                else if( pxCustomMetric->eType == eCustomMetricStringList )
+                {
+                    ulCharactersWritten = snprintf( pcCurrentWritePos,
+                                                    ulRemainingBufferLength,
+                                                    "\"%s\",",
+                                                    ( pxCustomMetric->xData.ppcStringList )[ i ] );
+                }
+                else if( pxCustomMetric->eType == eCustomMetricIpList )
+                {
+                    ulCharactersWritten = snprintf( pcCurrentWritePos,
+                                                    ulRemainingBufferLength,
+                                                    "\"%u.%u.%u.%u\",",
+                                                    ( unsigned int ) ( ( pxCustomMetric->xData.pulIpList )[ i ] >> 24 ) & 0xFF,
+                                                    ( unsigned int ) ( ( pxCustomMetric->xData.pulIpList )[ i ] >> 16 ) & 0xFF,
+                                                    ( unsigned int ) ( ( pxCustomMetric->xData.pulIpList )[ i ] >> 8 ) & 0xFF,
+                                                    ( unsigned int ) ( pxCustomMetric->xData.pulIpList )[ i ] & 0xFF );
+                }
+
+                if( !reportbuilderSNPRINTF_SUCCESS( ulCharactersWritten, ulRemainingBufferLength ) )
+                {
+                    eStatus = eReportBuilderBufferTooSmall;
+                    break;
+                }
+                else
+                {
+                    ulRemainingBufferLength -= ulCharactersWritten;
+                    pcCurrentWritePos += ulCharactersWritten;
+                }
+            }
+
+            if( eStatus != eReportBuilderSuccess )
+            {
+                break;
+            }
+
+            /* Discard the last comma. */
+            if( pxCustomMetric->ulLength > 0 )
+            {
+                pcCurrentWritePos -= 1;
+                ulRemainingBufferLength += 1;
+            }
+
+            /* Close the custom metric element */
+            ulCharactersWritten = snprintf( pcCurrentWritePos,
+                                            ulRemainingBufferLength,
+                                            "]}]," );
+
+            if( !reportbuilderSNPRINTF_SUCCESS( ulCharactersWritten, ulRemainingBufferLength ) )
+            {
+                eStatus = eReportBuilderBufferTooSmall;
+                break;
+            }
+            else
+            {
+                ulRemainingBufferLength -= ulCharactersWritten;
+                pcCurrentWritePos += ulCharactersWritten;
+            }
+        }
+    }
+
+    if( eStatus == eReportBuilderSuccess )
+    {
+        /* Discard the last comma. */
+        if( ulCustomMetricsArrayLength > 0 )
```

Comment:
As mentioned [earlier](https://github.com/FreeRTOS/FreeRTOS/pull/507/files#r577209450), this would be good to assert checked instead.